localstack/services/apigateway/helpers.py

```diff
@@ -828,6 +828,11 @@ def apply_json_patch_safe(subject, patch_operations, in_place=True, return_list=
                 path = operation["path"]
                 common.assign_to_path(subject, path, value={}, delimiter="/")
 
+            # minimum_compression_size is a unique path as it's a nullable integer,
+            # it would throw an error if it's an empty string
+            if operation["path"] == "/minimum_compression_size" and operation.get("value") == "":
+                operation["value"] = None
+
```

Comment:
I'm not sure I would add a specific case here in this helper. Wouldn't it be possible to fix the issue after the patch has been applied? So we don't have "subject" specific fixes in the global util? 

Something like the fix for `vpcEndpointIds` done after the patches in `update_rest_api`:
```python
        _patch_api_gateway_entity(rest_api, fixed_patch_ops)

        # fix data types after patches have been applied
        endpoint_configs = rest_api.endpoint_configuration or {}
        if isinstance(endpoint_configs.get("vpcEndpointIds"), str):
            endpoint_configs["vpcEndpointIds"] = [endpoint_configs["vpcEndpointIds"]]
        if rest_api.minimum_compression_size == "":
            rest_api.minimum_compression_size = None
```

I think this would be easier and would contain the fix where the issue lies, while keeping this util free of case specific logic.